@yongtang It looks like you need to fix the expected error message in your test:

Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu-python3/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-py3-opt/bin/bazel_pip/tensorflow/python/special_math_ops_test.runfiles/org_tensorflow/bazel_pip/tensorflow/python/ops/special_math_ops_test.py", line 251, in test_invalid_keyword_arguments
    invalid1="value1", invalid2="value2")
**AssertionError: "invalid keyword arguments for this function: invalid1, invalid2" does not match "invalid keyword arguments for this function: invalid2, invalid1"**

--- a/tensorflow/python/ops/special_math_ops.py
+++ b/tensorflow/python/ops/special_math_ops.py
@@ -154,8 +154,9 @@ def einsum(equation, *inputs, **kwargs):
   """
   name = kwargs.pop("name", None)
   if kwargs:
-    raise TypeError("invalid keyword arguments for this function: "
-        +", ".join([format(key) for key in reversed(list(kwargs.keys()))]))
+    raise TypeError("invalid keyword arguments for this function: " +
+                    ", ".join([format(key)
+                               for key in reversed(list(kwargs.keys()))]))
   with ops.name_scope(name, "einsum", [equation, inputs]) as name:
     if '...' in equation:
       raise ValueError('Subscripts with ellipses are not yet supported.')

--- a/tensorflow/python/ops/special_math_ops_test.py
+++ b/tensorflow/python/ops/special_math_ops_test.py
@@ -248,7 +248,7 @@ class EinsumTest(test.TestCase):
     with self.assertRaisesRegexp(TypeError,
         'invalid keyword arguments for this function: invalid1, invalid2'):
       _ = special_math_ops.einsum('ij,jk->ik', m0, m1, name="name",
-                                  invalid1="invalid1", invalid2="invalid2")
+                                  invalid1="value1", invalid2="value2")
 
   def test_dim_mismatch(self):
     for axes, input_shapes in self.dim_mismatch_cases:

--- a/tensorflow/tools/api/golden/tensorflow.pbtxt
+++ b/tensorflow/tools/api/golden/tensorflow.pbtxt
@@ -910,7 +910,7 @@ tf_module {
   }
   member_method {
     name: "einsum"
-    argspec: "args=[\'equation\'], varargs=inputs, keywords=None, defaults=None"
+    argspec: "args=[\'equation\'], varargs=inputs, keywords=kwargs, defaults=None"
   }
   member_method {
     name: "encode_base64"

